In [1]:
import pickle
import pandas as pd

# === Load trained pipeline ===
with open('random_forest_pipeline.pkl', 'rb') as file:
    model = pickle.load(file)

# === Define sample input ===
sample_input = {
    "Count": 1,
    "Country": "United States",
    "State": "California",
    "City": "Los Angeles",
    "Zip Code": 90001,
    "Latitude": 34.0522,
    "Longitude": -118.2437,
    "Gender": "Male",
    "Partner": "Yes",
    "Dependents": "No",
    "Tenure Months": 12,
    "Phone Service": "Yes",
    "Multiple Lines": "No",
    "Internet Service": "Fiber optic",
    "Online Security": "No",
    "Online Backup": "Yes",
    "Device Protection": "No",
    "Tech Support": "No",
    "Streaming TV": "Yes",
    "Streaming Movies": "No",
    "Contract": "Month-to-month",
    "Paperless Billing": "Yes",
    "Payment Method": "Electronic check",
    "Monthly Charges": 85.5,
    "Total Charges": 1020.5
}

# === Convert to DataFrame ===
input_df = pd.DataFrame([sample_input])

# === Run prediction ===
pred = model.predict(input_df)[0]
proba = model.predict_proba(input_df)[0][1]

# === Output ===
print(f"Predicted Churn Class: {pred}")
print(f"Churn Probability: {round(proba, 4)}")


Predicted Churn Class: 1
Churn Probability: 0.67


* Predicted Churn Class: 1 → Customer is likely to churn
* Churn Probability: 0.67 → 67% confidence